common/tools/dev-tool/src/commands/check.ts

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License
+
+import path from "node:path";
+import { resolveProject } from "../util/resolveProject";
+import { createPrinter } from "../util/printer";
+import { leafCommand } from "../framework/command";
+import { makeCommandInfo } from "../framework/command";
+import fs from "node:fs/promises";
+import { Check, isCheckFailedError } from "../framework/check";
+
+const log = createPrinter("check");
+
+export const commandInfo = makeCommandInfo("check", "run package checks", {
+  fix: {
+    kind: "boolean",
+    description: "attempt to fix failing checks, where supported",
+    default: false,
+  },
+  tag: {
+    kind: "string",
+    description: "run checks that match the given tag only. Available tags: 'release', 'ci'",
+  },
+  verbose: {
+    kind: "boolean",
+    shortName: "v",
+    description: "show output of check commands",
+    default: false,
+  },
+});
+
+export default leafCommand(commandInfo, async (options) => {
+  const checkFileNames = await fs.readdir(path.join(__dirname, "..", "checks"));
+
+  log.info("Running checks");
+
+  let checksRun = 0;
+  let checksPassed = 0;
+  let checksFailed = 0;
+  let warnings = 0;
+  let checksSkipped = 0;
+
+  for (const checkFile of checkFileNames) {
+    const checks = (await import(path.posix.join("../checks", checkFile))) as Record<string, Check>;
+    log(checkFile);
+    for (const [exportName, check] of Object.entries(checks)) {
+      if (options.tag) {
+        // --tag specified, run checks which match that tag (including checks with wildcard "*")
+        if (!check.tags) {
+          continue;
+        }
+
+        if (
+          check.tags !== "*" &&
+          !check.tags.some((x) => x.toLowerCase() === options.tag?.toLowerCase())
+        ) {
+          continue;
+        }
+      } else {
+        // --tag not specified, only run checks which don't have a tag or have tag "*"
+        if (check.tags && check.tags !== "*") {
+          continue;
+        }
+      }
+
+      const name = check.name ?? exportName;
+      try {
+        const project = await resolveProject();
+
+        if (!check.enable || (await check.enable?.(project))) {
+          ++checksRun;
+          await check.check({
+            fix: !!check.hasFix && options.fix,
+            verbose: options.verbose,
+            project: await resolveProject(),
```

Comment:
reuse the resolved one above?
```suggestion
            project,
```

Reply:
I made it resolve the project every time here on purpose since each check might make changes to things like the package.json. I think if we didn't do it every time it might get stale

Reply:
wait duh I see what you mean, we're doing it in the loop anyway so only resolving once per check. I'll update